config/v1/types_infrastructure.go

```diff
@@ -531,9 +531,296 @@ type OvirtPlatformStatus struct {
+	// There is three valid options: unset, Allowed and NotAllowed.
+	// +kubebuilder:validation:Optional
+	ControlPlane VSpherePlatformDeploymentSuitable `json:"controlPlane,omitempty"`
+
+	// PlacementConstraint encapsulates the placement constraints
+	// used within this deployment zone.
+	// +kubebuilder:validation:Optional
+	PlacementConstraint VSpherePlatformPlacementConstraint `json:"placementConstraint"`
+}
+
+// VSpherePlatformNodeNetworkingSpec holds the network CIDR(s) and port group name for
+// including and excluding IP ranges in the cloud provider.
+// This would be used for example when multiple network adapters are attached to
+// a guest to help determine which IP address the cloud config manager should use
+// for the external and internal node networking.
+type VSpherePlatformNodeNetworkingSpec struct {
+	// networkSubnetCidr IP address on VirtualMachine's network interfaces included in the fields' CIDRs
+	// that will be used in respective status.addresses fields.
+	// +kubebuilder:validation:Optional
+	NetworkSubnetCIDR []string `json:"networkSubnetCidr,omitempty"`
+
+	// network VirtualMachine's VM Network names that will be used to when searching
+	// for status.addresses fields. Note that if InternalNetworkSubnetCIDR and
+	// ExternalNetworkSubnetCIDR are not set, then the vNIC associated to this network must
+	// only have a single IP address assigned to it.
+	// +kubebuilder:validation:Optional
+	Network string `json:"network,omitempty"`
+
+	// excludeNetworkSubnetCidr IP addresses in subnet ranges will be excluded when selecting
+	// the IP address from the VirtualMachine's VM for use in the status.addresses fields.
+	// +kubebuilder:validation:Optional
+	ExcludeNetworkSubnetCIDR []string `json:"excludeNetworkSubnetCidr,omitempty"`
+}
+
+// VSpherePlatformNodeNetworking holds the external and internal node networking spec.
+type VSpherePlatformNodeNetworking struct {
+	// external represents the VSpherePlatformNodeNetworkingSpec of the node that is externally routable.
+	// +kubebuilder:validation:Optional
+	External VSpherePlatformNodeNetworkingSpec `json:"external"`
+	// internal represents the VSpherePlatformNodeNetworkingSpec of the node that is routable only within the cluster.
+	// +kubebuilder:validation:Optional
+	Internal VSpherePlatformNodeNetworkingSpec `json:"internal"`
+}
+
 // VSpherePlatformSpec holds the desired state of the vSphere infrastructure provider.
-// This only includes fields that can be modified in the cluster.
-type VSpherePlatformSpec struct{}
+// In the future the cloud provider operator, storage operator and machine operator will
+// use these fields for configuration.
+type VSpherePlatformSpec struct {
+	// vcenters holds the connection details for services to communicate with vCenter.
+	// Currently, only a single vCenter is supported.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:MaxItems=1
+	// +kubebuilder:validation:MinItems=1
```

Comment:
I'm not sure optional and min/max 1 are coherent

Reply:
Changing minitems to 0